### PR TITLE
Remove cloud config check in VirtualTopologyGroupService

### PR DESCRIPTION
**Pull request by qqu0127**
Remove the dependency on CloudConfig for setting virtual topology group.

### Issues

- [ ] My PR addresses the following Helix issues and references them in the PR description:

(#200 - Link your issue number here: You can write "Fixes #XXX". Please use the proper keyword so that the issue gets closed automatically. See https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue
Any of the following keywords can be used: close, closes, closed, fix, fixes, fixed, resolve, resolves, resolved)

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This PR removes the cloud config check in `VirtualTopologyGroupService` since we want to generalize this feature to all environments, both cloud and on-prem. 

Now the validation includes:
1. Request param validation
2. Cluster has topology aware enabled
3. Number of virtual zones should be less than the number of instances
4. In maintenance mode if configured

### Tests

- [X] The following tests are written for this issue:

Modified `TestVirtualTopologyGroupService` 

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java [hunk 1]
```diff
@@ -89,11 +89,6 @@ public VirtualTopologyGroupService(HelixAdmin helixAdmin, ClusterService cluster
    */
   public void addVirtualTopologyGroup(String clusterName, Map<String, String> customFields) {
     // validation
-    CloudConfig cloudConfig = _configAccessor.getCloudConfig(clusterName);
```
Comment (desaikomal) [inline helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java:92]: What if someone wants to use it in Cloud environment. Can we just allow the config to be NULL and still proceed???

Comment (qqu0127) [reply helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java:92]: It's totally fine to run in cloud environment. It doesn't matter whether its null or not. CloudConfig isn't used anywhere else after the validation.

Comment (desaikomal) [reply helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java:92]: So it is like feature-flag kind of behavior.

Comment (qqu0127) [reply helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java:92]: Not really, virtual group API was first only supposed to run in cloud environment, so `cloudConfig` was a prerequisite, now it's removed. There is no information or dependency needed from cloudConfig. A closer thing to feature-flag is `clusterConfig.isTopologyAwareEnabled`.

Comment (desaikomal) [reply helix-rest/src/main/java/org/apache/helix/rest/server/service/VirtualTopologyGroupService.java:92]: thanks for explaining.

**Review by desaikomal — APPROVED** [review]
LGTM

**Comment by qqu0127** [comment]
This PR is ready to merge, approved by @desaikomal , the failed test is flaky and is a known issue.
Commit message:
Remove the dependency on CloudConfig for setting virtual topology group.

**Pull request merged**